### PR TITLE
Implementing Redelegation

### PR DESCRIPTION
**Pull request by majecty**
This PR closes https://github.com/CodeChain-io/codechain/issues/1740

### CODE REVIEW

File: core/src/consensus/stake/mod.rs [hunk 1]
```diff
@@ -226,6 +231,33 @@ fn revoke(state: &mut TopLevelState, fee_payer: &Address, delegatee: &Address, q
...
 }
 
+fn redelegate(
+    state: &mut TopLevelState,
+    fee_payer: &Address,
+    prev_delegatee: &Address,
+    next_delegatee: &Address,
+    quantity: u64,
+) -> StateResult<()> {
+    let delegator = StakeAccount::load_from_state(state, fee_payer)?;
```
Comment (foriequal0) [inline core/src/consensus/stake/mod.rs]: Check `next_delegatee` is valid candidate like `delegate`
https://github.com/CodeChain-io/codechain/pull/1742/files#diff-a25f9a1ca8af0d023966594384adb0f4R194

Comment (majecty) [reply core/src/consensus/stake/mod.rs]: Thanks. I missed it.

**Comment by majecty** [comment]
@foriequal0 I fixed it.

**Comment by foriequal0** [comment]
Can you add some tests?
1. Can redelegate
2. Fail on too much quantity
3. Fail on next_candidate jailed/banned

**Pull request merged**